docs/types.rst

```diff
@@ -0,0 +1,23 @@
+.. currentmodule:: jsonschema
+
+=============
+Type Checking
+=============
+
+Each :class:`IValidator` has an associated :class:`TypeChecker`. The
+TypeChecker provides an immutable mapping between names of types and
+functions that can test if an instance is of that type. The defaults are
+suitable for most users - each of the predefined Validators (Draft3, Draft4)
+has a :class:`TypeChecker` that can correctly handle that draft.
+
+See :ref:`validating-types` for an example of providing a custom type check.
+
+.. autoclass:: TypeChecker
+    :members:
+
+.. autoexception:: jsonschema.exceptions.UndefinedTypeCheck
+
+    Raised when trying to remove a type check that is not known to this
+    TypeChecker. Internally this is also raised when calling
```

Comment:
We don't really get a choice on the "internally" I think.

`TypeChecker.is_type` is public, so everyone needs to know it raises this exception there too, so we can never really consider it internal anymore.